### DISCUSSION TITLE
XMLSchedulingDataProcessorPlugin XML job data not scanning properly

### DISCUSSION THREAD

**Discussion by ghost** [quartznet/quartznet#1469] [General]
Hi Team,

I have used XMLSchedulingDataProcessorPlugin for an automatic refresh for my job data. Its working in the sample application but the file change is not detected within my application code. I can see the XMLplugin CRON is working as configured in the scan interval. The only difference is I am using a custom provide for JobStore.

I have used the code as below. Its seems simple as that but something weired. 
----------------------------------
IScheduler quartzScheduler = await SchedulerBuilder.Create(properties)
   .UseDefaultThreadPool(x => x.MaxConcurrency = 5)           
   .UsePersistentStore(x =>
   {                   
       x.UseProperties = true;
       //x.UseClustering();
       x.UseJsonSerializer();                 
   })
   .UseXmlSchedulingConfiguration(x =>
   {
       x.Files = new[] { "~/quartz_jobs.xml" };
       x.FailOnFileNotFound = true;
       x.FailOnSchedulingError = true;                   
       x.ScanInterval = TimeSpan.FromSeconds(30);
   })
   .BuildScheduler();

await quartzScheduler.Start();
------------------------

**Comment by lahma** [comment]
You should check the logs on debug level and see what decisions are being made.

**Reply by ghost** [reply]
I can see the debug log and its always shows as file is unchanged. Strange

---------------------
12/Dec/2021 08:17:29 AM [106] [DEBUG] MoveNext - Calling Execute on job XMLSchedulingDataProcessorPlugin.XMLSchedulingDataProcessorPlugin_xml_quartz_jobs_xml
12/Dec/2021 08:17:29 AM [106] [DEBUG] MoveNext - File 'C:\wwwroot\sample\WebApplication\quartz_jobs.xml' unchanged.
--------------------------
I have provided full permission for IIS_USER for this particular file as well

**Reply by ghost** [reply]
Could you please give any clue on this?

Thanks,
Jineesh

**Reply by lahma** [reply]
Sorry, can't help here as it's your environment. Maybe check timestamps etc what could cause file not to be seen as changed.

**Reply by ghost** [reply]
The timestamp is changed. And the latest file will fetch on the next rerun of the applciation. But the scanning is not working as expected.
